rusk/tests/services/conversion.rs

```diff
@@ -0,0 +1,209 @@
+        block_height,
+        BLOCK_GAS_LIMIT,
+        vec![],
+        None,
+    )
+    .expect("The generator procedure should succeed");
+
+    let tx = txs.first().expect("tx to be processed");
+    let gas_spent = tx.gas_spent;
+
+    let phoenix_balance = wallet
+        .get_balance(0)
+        .expect("Getting phoenix balance should succeed");
+    let moonlight_account = wallet
+        .get_account(0)
+        .expect("Getting account data should succeed");
+
+    assert_eq!(
+        phoenix_balance.value, INITIAL_PHOENIX_BALANCE - CONVERT_VALUE - gas_spent,
+        "The Phoenix notes must be of its initial value minus the converted amount and gas spent"
+    );
+    assert_eq!(
+        moonlight_account.balance, INITIAL_MOONLIGHT_BALANCE + CONVERT_VALUE,
+        "The Moonlight account should have its initial balance plus the converted amount"
+    );
+}
+
+/// Makes a transaction that converts Dusk from Phoenix to Moonlight, and
+/// produces a block with a single transaction, checking balances accordingly.
+fn wallet_convert_to_phoenix(
+    rusk: &Rusk,
+    wallet: &wallet::Wallet<TestStore, TestStateClient>,
+    block_height: u64,
+) {
+    const CONVERT_VALUE: u64 = INITIAL_PHOENIX_BALANCE / 2;
+    const GAS_LIMIT: u64 = 1_000_000_000;
+
+    let moonlight_account = wallet
+        .get_account(0)
+        .expect("Getting account data should succeed");
+    let phoenix_balance = wallet
+        .get_balance(0)
+        .expect("Getting phoenix balance should succeed");
+
+    assert_eq!(
+        moonlight_account.balance, INITIAL_MOONLIGHT_BALANCE,
+        "The Moonlight account should have its initial balance"
+    );
+    assert_eq!(
+        phoenix_balance.value, INITIAL_PHOENIX_BALANCE,
+        "The Phoenix notes must be of its initial value"
+    );
+
+    let mut rng = StdRng::seed_from_u64(0xdead);
+
+    let tx = wallet
+        .moonlight_to_phoenix(&mut rng, 0, 0, CONVERT_VALUE, GAS_LIMIT, 1)
+        .expect("Creating conversion transaction should succeed");
+
+    let txs: Vec<SpentTransaction> = generator_procedure(
+        rusk,
+        &[tx],
+        block_height,
+        BLOCK_GAS_LIMIT,
+        vec![],
+        None,
+    )
+    .expect("The generator procedure should succeed");
+
+    let tx = txs.first().expect("tx to be processed");
+    let gas_spent = tx.gas_spent;
+
+    let moonlight_account = wallet
+        .get_account(0)
+        .expect("Getting account data should succeed");
+    let phoenix_balance = wallet
+        .get_balance(0)
+        .expect("Getting phoenix balance should succeed");
+
+    assert_eq!(
+        moonlight_account.balance, INITIAL_MOONLIGHT_BALANCE - CONVERT_VALUE - gas_spent,
+        "The Moonlight account must have its initial value minus the converted amount and gas spent"
+    );
+    assert_eq!(
+        phoenix_balance.value, INITIAL_PHOENIX_BALANCE + CONVERT_VALUE,
+        "The Phoenix notes should must be of their value plus the converted amount"
```

Comment:
"The Phoenix notes should have its initial balance plus the converted amount" or something similar

Reply:
Fixed